### ISSUE TITLE
conda info --json and package lookup

### ISSUE THREAD

**Issue by alanhdu** [conda/conda#2444]
If you set the `--json` flag for `conda info` when searching for packages, you sometimes get nothing:

```bash
$ conda info numpy=1.11.0=py35_0
Fetching package metadata: ....

numpy 1.11.0 py35_0
-------------------
file name   : numpy-1.11.0-py35_0.tar.bz2
name        : numpy
version     : 1.11.0
build number: 0
build string: py35_0
channel     : defaults
size        : 6.1 MB
date        : 2016-03-28
license     : BSD
md5         : 1900998c19c5e310687013f95374bba2
installed environments:
dependencies:
    mkl 11.3.1
    python 3.5*
$ conda info --json numpy=1.11.0=py35_0
{}
```

Things work fine for `conda info --json numpy`, so it's something with the spec format.

conda info:
```
             platform : linux-64
        conda version : 4.0.5
  conda-build version : not installed
       python version : 2.7.11.final.0
     requests version : 2.9.1
     root environment : /opt/conda  (writable)
  default environment : /opt/conda
     envs directories : /opt/conda/envs
        package cache : /opt/conda/pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
    is foreign system : False
```

**Comment by alanhdu** [comment]
I wrote a short fix (#2445), but we could also just deprecate the `conda info <package>` search anyways. It seems pretty redundant given `conda search` (e.g. just use `conda search --json --spec numpy=1.11.0=py35_0` instead).

**Comment by alanhdu** [comment]
On a slightly broader point, I don't think the `--json` flag really plays well with any of the other flags in `conda info`. `conda info --json --license`, `conda info --json`, and `conda info --json --envs`all return the same thing (for example), while `conda info --json --all` gives you:

```bash
$ conda info --json --all
Current conda install:

             platform : linux-64
        conda version : 4.0.5
  conda-build version : not installed
       python version : 2.7.11.final.0
     requests version : 2.9.1
     root environment : /opt/conda  (writable)
  default environment : /opt/conda
     envs directories : /opt/conda/envs
        package cache : /opt/conda/pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
    is foreign system : False

{
  "_channels": "https://repo.continuum.io/pkgs/free/linux-64/\n                        https://repo.continuum.io/pkgs/free/noarch/\n                        https://repo.continuum.io/pkgs/pro/linux-64/\n                        https://repo.continuum.io/pkgs/pro/noarch/", 
  "_envs_dirs": "/opt/conda/envs", 
  "_pkgs_dirs": "/opt/conda/pkgs", 
  "_rtwro": "writable", 
  "channels": [
    "https://repo.continuum.io/pkgs/free/linux-64/", 
    "https://repo.continuum.io/pkgs/free/noarch/", 
    "https://repo.continuum.io/pkgs/pro/linux-64/", 
    "https://repo.continuum.io/pkgs/pro/noarch/"
  ], 
  "conda_build_version": "not installed", 
  "conda_version": "4.0.5", 
  "default_prefix": "/opt/conda", 
  "envs": [], 
  "envs_dirs": [
    "/opt/conda/envs"
  ], 
  "is_foreign": false, 
  "pkgs_dirs": [
    "/opt/conda/pkgs"
  ], 
  "platform": "linux-64", 
  "python_version": "2.7.11.final.0", 
  "rc_path": null, 
  "requests_version": "2.9.1", 
  "root_prefix": "/opt/conda", 
  "root_writable": true, 
  "sys_rc_path": "/opt/conda/.condarc", 
  "user_rc_path": "/.condarc"
}
```

So, I guess we can either (1) support all combinations of `--json` and the other flags, (2) raise some informative errors that `--json` is incompatible with other flags (better than silently failing) or (3) use subcommands instead of flags (effectively does (2), but it shouldn't need custom validation logic).
